### ISSUE TITLE
Regression: calling .register on typing.KeysView subclass causes false positive error `Member "register" is unknown`

### ISSUE THREAD

**Issue by jab** [microsoft/pyright#4241]
**Describe the bug**

pyright v1.1.281 incorrectly emits the error `Member "register" is unknown` for the following line of code:
https://github.com/jab/bidict/blob/522c9b69249d66c201fed608f95a5da8d3064ed8/bidict/_base.py#L51

pyright v1.1.278 does not emit an error for this line, as expected.

Copy/pasted excerpt:
```python
import typing as t

from ._typing import KT  # a typevar


class BidictKeysView(t.KeysView[KT], t.ValuesView[KT]):
    """Since the keys of a bidict are the values of its inverse (and vice versa),
    the :class:`~collections.abc.ValuesView` result of calling *bi.values()*
    is also a :class:`~collections.abc.KeysView` of *bi.inverse*.
    """


dict_keys: t.Type[t.KeysView[t.Any]] = type({}.keys())
BidictKeysView.register(dict_keys)  # <-- false positive error emitted here
```

**Comment by erictraut** [comment]
This is due to a recent change in the typeshed stubs, not a bug in pyright. `KeysView` inherits from `Sized`, and `Sized` was previously decleared in `typing.pyi` as having a metaclass of `ABCMeta`. In recent versions of `typing.pyi`, this metaclass association was removed. If you think this is incorrect, please file a bug in the typeshed project.

**Comment by jab** [comment]
Sorry for reporting here and thank you for the pointer! Followed up in https://github.com/python/typeshed/issues/9296.

**Comment by jab** [comment]
To follow back up here, the typeshed maintainers agreed this was a regression and have reverted the associated changes.

**Comment by erictraut** [comment]
@jab, thanks. I just pulled the latest typeshed stubs into pyright, so this will be addressed in the next pyright release.

**Comment by jab** [comment]
Great, thank you!
